### ISSUE TITLE
Cannot copy from folders with dots in the name

### ISSUE THREAD

**Issue by tothi** [moshfeu/vscode-compare-folders#136]
Let's say we have two folders:

My Folder: `/home/istvan/w/test`
Compared Folder: `/home/istvan/w/test.2`

Here is `hello.txt` only in the compared folder:
`/home/istvan/w/test.2/hello.txt`

If we try to "Copy to my folder" (using right click context menu on hello.txt), it gives the following js error at `workbench.desktop.main.js:149` (stating wrong path expansion `/home/istvan/w/test.2/.2/hello.txt` instead of `/home/istvan/w/test.2/hello.txt`):

```
Error: ENOENT: no such file or directory, stat '/home/istvan/w/test.2/.2/hello.txt'
	at statSync (node:fs:1615:3)
	at t.statSync (node:electron/js2c/asar_bundle:2:4557)
	at Object.statSync (/home/istvan/.vscode-oss/extensions/moshfeu.compare-folders-0.23.0-universal/node_modules/graceful-fs/polyfills.js:318:34)
	at getStatsSync (/home/istvan/.vscode-oss/extensions/moshfeu.compare-folders-0.23.0-universal/node_modules/fs-extra/lib/util/stat.js:58:18)
	at Object.checkPathsSync (/home/istvan/.vscode-oss/extensions/moshfeu.compare-folders-0.23.0-universal/node_modules/fs-extra/lib/util/stat.js:90:33)
	at copySync (/home/istvan/.vscode-oss/extensions/moshfeu.compare-folders-0.23.0-universal/node_modules/fs-extra/lib/copy-sync/copy-sync.js:24:38)
	at CompareFoldersProvider.copyToFolder (/home/istvan/.vscode-oss/extensions/moshfeu.compare-folders-0.23.0-universal/out/providers/foldersCompareProvider.js:222:37)
	at CompareFoldersProvider.copyToMy (/home/istvan/.vscode-oss/extensions/moshfeu.compare-folders-0.23.0-universal/out/providers/foldersCompareProvider.js:129:18)
	at d.h (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:141:140094)
	at d.$executeContributedCommand (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:141:140954)
	at l.S (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:150:10829)
	at l.Q (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:150:10595)
	at l.M (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:150:9685)
	at l.L (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:150:8903)
	at f.value (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:150:7567)
	at n.y (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
	at n.fire (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
	at o.fire (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:103:14039)
	at f.value (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:171:7903)
	at n.y (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:80:1902)
	at n.fire (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:80:2119)
	at o.fire (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:103:14039)
	at MessagePortMain.<anonymous> (/usr/lib/code/out/vs/workbench/api/node/extensionHostProcess.js:171:6183)
	at MessagePortMain.emit (node:events:513:28)
	at MessagePortMain._internalPort.emit (node:electron/js2c/utility_init:2:367)
```

Using Code - OSS 1.85.1 on Linux (Node.js: 18.15.0), with Compare Folders v0.23.0 (from marketplace).

**Comment by moshfeu** [comment]
Thanks @tothi  for reporting this. I'll look into it soon

**Comment by moshfeu** [comment]
Was able to validate. I'll try to fix it soon

**Comment by moshfeu** [comment]
Related to https://github.com/nodejs/node/issues/6229

**Comment by moshfeu** [comment]
Can you try this version please?

[compare-folders-0.24.2.vsix.zip](https://github.com/moshfeu/vscode-compare-folders/files/13850364/compare-folders-0.24.2.vsix.zip)

[Install extension from vsix file](https://code.visualstudio.com/docs/editor/extension-marketplace#_install-from-a-vsix)

**Comment by tothi** [comment]
cannot. when I add the VSIX it hangs in "activating..." state and the extension is not fully available.

**Comment by moshfeu** [comment]
> cannot. when I add the VSIX it hangs in "activating..." state and the extension is not fully available.

You're right. An error thrown on linux. Checking

**Comment by moshfeu** [comment]
Can you try the latest insider version?
https://marketplace.visualstudio.com/items?itemName=moshfeu.compare-folders-insider
You need to disable the regular version first

Sorry for the inconvenient, there is an issue on linux when building the extension on mac. Normally it goes through the ci which build the extension on Linux but since it's an internal version, I built it on my mac computer. Anyway.. I built it now on a linux machine so I hope it will work for you this time 🤞

**Comment by NordiskaRobin** [comment]
I had the same issue and i can verify that with the insider version it works now! :)

**Comment by moshfeu** [comment]
Thank you @NordiskaRobin!
I'll push the version to the regular extension soon

**Comment by moshfeu** [comment]
Published in version 0.24.2
